### ISSUE TITLE
Feature idea: Select chars by regex, get text and bounding box in return

### ISSUE THREAD

**Issue by jsvine** [jsvine/pdfplumber#201]
It could be helpful to add a `Page` method that worked something like this:

```python
p = pdf.pages[0]
phrases = p.find_text(r"Updated on \d{4}-\d{2}-\d{2}")
```

... where `phrases` is something like:

```python
[
  {
    "text": "Updated on 2020-04-07",
    "x0": Decimal(119.39),
    "top": Decimal(50.10),
    [...]
  },
  {
    "text": "Updated on 2020-04-04",
    "x0": Decimal(119.39),
    "top": Decimal(80.10),
    [...]
  }
]
```

**Comment by zypriafl** [comment]
Hi Jeremy!

I'm the Co-Founder of Konfuzio.com, among other functionality we offer exactly this method in our Python Package. E. g. you can literally draw bounding boxes with regex, or even train Regex using our DeepLearing Backend without writing them, just by selecting the text you want to extract.

![image](https://user-images.githubusercontent.com/6725189/78759042-527db900-797f-11ea-9139-37ea8b3be268.png)

Normally we are providing our offerings to enterprise clients, however, feel free to ping us at info(-at-)konfuzio.com. Perhaps we can support you to publish your great content :)

**Comment by jsfenfen** [comment]
Plus one! This is functionally how I use this a lot (but with pandas selectors, which doesn't make regexes simple). E.g. ignore everything below where it says "Total".

**Comment by dumbPy** [comment]
matching with pdfminer's `LTTextLine.get_text()` and returning their LTChars, and then converting them to whatever you desire is an easy solution.
I will start a PR for this and we can discuss it further. Let me know @jsvine

**Comment by anirudhpnbb** [comment]
Maybe it(the page) can be temporarily stored as an image, then extract results I mean the bounding boxes and then can be deleted right?

**Comment by jsvine** [comment]
Just landed in v0.7.0!
